soapui/src/main/java/com/smartbear/integrations/swaggerhub/ImportFromHubDialog.java

```diff
@@ -0,0 +1,314 @@
+            progressIndicator.setMaxWidth(table.getWidth() * 0.7);
+            Platform.runLater(() -> {
+                stackPane.getChildren().add(progressIndicator);
+                table.clearTable();
+            });
+            boolean importPrivate = false;
+            String apiKey = apiKeyField.getText();
+            String searchQuery = searchField.getText();
+
+            String uri;
+
+            if (rememberCombo.isSelected()) {
+                Workspace workspace = SoapUI.getWorkspace();
+                workspace.getSettings().setString(SWAGGER_HUB_API_KEY, apiKeyField.getText());
+            }
+
+            if (StringUtils.isNotEmpty(apiKey)) {
+                if (searchInMyHub.isSelected()) {
+                    uri = SWAGGERHUB_API + "?filter=user";
+                } else {
+                    uri = SWAGGERHUB_API + "?limit=" + SEARCH_LIMIT;
+                }
+                importPrivate = true;
+            } else {
+                uri = SWAGGERHUB_API + "?limit=" + SEARCH_LIMIT;
+            }
+
+            if (StringUtils.isNotEmpty(searchQuery)) {
+                try {
+                    uri += "&query=" + URLEncoder.encode(searchQuery.trim(), "UTF-8");
+                } catch (UnsupportedEncodingException e) {
+                    log.error(e.getMessage(), e);
+                }
+            }
+
+            try {
+                HttpGet get = new HttpGet(uri);
+                if (importPrivate) {
+                    get.setHeader("Authorization", apiKey);
+                }
+                HttpResponse response = HttpClientSupport.getHttpClient().execute(get);
+
+                List<ApiDescriptor> descriptors = new ApisJsonImporter().importApis(
+                        new String(ByteStreams.toByteArray(response.getEntity().getContent())));
+
+                Platform.runLater(() -> {
+                    for (ApiDescriptor descriptor : descriptors) {
+                        table.addDescription(descriptor);
+                    }
+                });
+            } catch (Exception e) {
+                log.error(GETTING_LIST_OF_DEFINITIONS_ERROR, e);
+                Platform.runLater(() -> buildAlert(GETTING_LIST_OF_DEFINITIONS_ERROR, "Error", ERROR).showAndWait());
+            } finally {
+                Platform.runLater(() -> {
+                    stackPane.getChildren().remove(progressIndicator);
+                    searchButton.setDisable(false);
+                });
+            }
+        });
+    }
+
+    private List<RestService> importApis() {
+        List<RestService> result = new ArrayList<>();
+        try {
+            SwaggerHubAPITableModel model = (SwaggerHubAPITableModel) table.getSelectionModel().getSelectedItem();
+            ApiDescriptor descriptor = model.getDescriptor();
+            String defaultVersionUrl = descriptor.swaggerUrl;
+
+            SwaggerImporter importer = new Swagger2Importer(project, "application/json");
```

Comment:
Do you guarantee that users cannot select APIs of other versions?

Reply:
Of course he can choose other API version and it is an issue.
We will try to fix as much as we can.